File: CookIt.API/Dtos/RecipeForListDto.cs
using CookIt.API.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CookIt.API.Dtos
{
    public class RecipeForListDto
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public Host Host { get; set; }
        public Uri Url { get; set; }
        public Uri ImageUrl { get; set; }
        public List<Ingredient> Ingredients { get; set; }
        public List<Ingredient> MatchedIngredients { get; set; }
    }

}

File: CookIt-Angular/src/app/core/models/recipeForListDto.ts
import { IngredientWithIsMatchDto } from './ingredientWithIsMatchDto';
import { Host } from './host';

export class RecipeForListDto {
    id: string;
    title: string;
    host: Host;
    url: string;
    imageUrl: string;
    imageSrc: string;
    isFavorite: boolean;
    ingredients: IngredientWithIsMatchDto[];
    matchedIngredientsCount: number;
    constructor(id: string, title: string, host: Host, url: string, imageUrl: string, imageSrc: string, isFavorite: boolean, ingredients: IngredientWithIsMatchDto[], matchedIngredientsCount: number) {
        this.id = id;
        this.title = title;
        this.host = host;
        this.url = url;
        this.imageUrl = imageUrl;
        this.imageSrc = imageSrc;
        this.isFavorite = isFavorite;
        this.ingredients = ingredients;
        this.matchedIngredientsCount = matchedIngredientsCount;
    }
};


